### PR TITLE
Add lint test with tox flake8

### PR DESCRIPTION
**Pull request by sky-joker**
##### SUMMARY

The current CI hasn't been able to the detection of unused variables and imported unused libraries.  
This PR is to add to able to the detection of unused variables and imported unused libraries with tox and flake8.  
Also, fix to syntax errors some modules and libraries.

fixes: https://github.com/ansible-collections/community.zabbix/issues/130

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/base.py
plugins/module_utils/helpers.py
plugins/module_utils/wrappers.py
plugins/modules/zabbix_action.py
plugins/modules/zabbix_group.py
plugins/modules/zabbix_group_info.py
plugins/modules/zabbix_maintenance.py
plugins/modules/zabbix_mediatype.py
plugins/modules/zabbix_template_info.py
plugins/modules/zabbix_user.py
plugins/modules/zabbix_valuemap.py
plugins/modules/zabbix_service.py
.github/workflows/plugins.yml
tox.ini

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by D3DeFi** [comment]
thank you for this contribution @sky-joker and for your work. Lets rebase this after #142 is merged

File: plugins/modules/zabbix_maintenance.py [hunk 1]
```diff
@@ -350,12 +350,12 @@ def main():
...
-            str(maintenance_type) != maintenance["maintenance_type"] or
-            str(int(start_time)) != maintenance["active_since"] or
-            str(int(start_time + period)) != maintenance["active_till"] or
-            str(desc) != maintenance['description']
+            sorted(group_ids) != sorted(maintenance["groupids"])
+            or sorted(host_ids) != sorted(maintenance["hostids"])
+            or str(maintenance_type) != maintenance["maintenance_type"]
+            or str(int(start_time)) != maintenance["active_since"]
+            or str(int(start_time + period)) != maintenance["active_till"]
+            or str(desc) != maintenance['description']
```
Comment (D3DeFi) [inline plugins/modules/zabbix_maintenance.py:358]: isn't this against PEP8 or something?

File: plugins/modules/zabbix_service.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def main():
     service = Service(module)
     service_ids = service.get_service_ids(name)
     if service_ids:
-        service_json = service.dump_services(service_ids)
+        service.dump_services(service_ids)
```
Comment (D3DeFi) [inline plugins/modules/zabbix_service.py:231]: this whole function call is probably uneccessary

**Comment by sky-joker** [comment]
Thank you @D3DeFi  for the looking this PR :)  
If possible, I want to increase the priority of this PR merge.  
The refactoring process is currently underway, but I would like to assume that the refactoring PR will be subject to syntax testing with flake8.
I'd like to eliminate the possibility of doing a flake8 error correction after refactoring.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+commands =
+  flake8 plugins {posargs}
+[testenv:venv]
+# This env is used to push new release on Galaxy
+install_command = pip install {opts} {packages}
+deps = ansible
+commands = {posargs}
+
+[flake8]
+# E123, E125 skipped as they are invalid PEP-8.
```
Comment (D3DeFi) [inline tox.ini:23]: Just to confirm, is this because PEP8 recognizes both E123 styles as a valid, but flake8 does not?

**Comment by D3DeFi** [comment]
> 
> 
> Thank you @D3DeFi for the looking this PR :)
> If possible, I want to increase the priority of this PR merge.
> The refactoring process is currently underway, but I would like to assume that the refactoring PR will be subject to syntax testing with flake8.
> I'd like to eliminate the possibility of doing a flake8 error correction after refactoring.

yes, makes sense. Lets merge this asap

File: plugins/modules/zabbix_maintenance.py (hunk 1 above)
Comment (sky-joker) [reply plugins/modules/zabbix_maintenance.py:358]: W503 is currently deprecated.
see: https://www.flake8rules.com/rules/W503.html

W504 is currently recommended.
see: https://www.flake8rules.com/rules/W504.html

**Comment by D3DeFi** [comment]
@sky-joker can I merge then?

File: tox.ini (hunk 3 above)
Comment (sky-joker) [reply tox.ini:23]: That's right :)

File: plugins/modules/zabbix_service.py (hunk 2 above)
Comment (sky-joker) [reply plugins/modules/zabbix_service.py:231]: I’m going to check this function call necessary or unnecessary after the merge.

**Comment by sky-joker** [comment]
> @sky-joker can I merge then?

Yes, please merge!

**Comment by D3DeFi** [comment]
Thank you!

File: tox.ini (hunk 3 above)
Comment (sky-joker) [reply tox.ini:23]: PEP8 does not detect E123, so flake8 skips it.

**Comment by sky-joker** [comment]
Thank you @D3DeFi :)

File: tox.ini (hunk 3 above)
Comment (sky-joker) [reply tox.ini:23]: FYI
Other collections tox settings are also skipping E123.

https://github.com/search?q=org%3Aansible-collections+%22invalid+PEP-8%22&type=Code

**Pull request merged**